src/test/ui/async-await/return-ty-unsize-coercion.rs

```diff
@@ -0,0 +1,34 @@
+// Check that we apply unsizing coercions based on the return type.
```

Comment:
Can you also include the string slice based regression test?

Reply:
I'm not sure what test you have in mind exactly? 

Note that this errors in both sequential *and* async functions:

```rust

// Unsizing coercion from `Box<String>` to `Box<str>`.
fn unsize_string_coercion() {
    fn sync_example() -> Box<str> {
        Box::new("".to_string())
    }

    async fn async_example() -> Box<str> {
        Box::new("".to_string())
    }
}
```

Reply:
This one: https://github.com/rust-lang/rust/issues/60424#issuecomment-528938362